nsdl-c/sn_nsdl_lib.h

```diff
@@ -156,52 +158,36 @@ typedef enum sn_nsdl_resource_mode_ {
 } sn_nsdl_resource_mode_e;
 
 /**
- * \brief Resource registration parameters
+ * \brief Defines dynamic parameters for the resource.
  */
-typedef struct sn_nsdl_resource_parameters_ {
-    unsigned int     observable:2;
-    unsigned int     registered:2;
-
-    uint16_t    resource_type_len;
-    uint16_t    interface_description_len;
-
-    uint16_t    coap_content_type;
-//    uint8_t     mime_content_type;
-
-    uint8_t     *resource_type_ptr;
-    uint8_t     *interface_description_ptr;
-
-} sn_nsdl_resource_parameters_s;
+typedef struct sn_nsdl_dynamic_resource_parameters_ {
+    uint8_t          registered:2;
+    uint8_t          (*sn_grs_dyn_res_callback)(struct nsdl_s *, sn_coap_hdr_s *, sn_nsdl_addr_s *, sn_nsdl_capab_e);
+    uint8_t          external_memory_block;
+    uint8_t          access:4;
+    uint16_t         coap_content_type; // uint8_t?
+} sn_nsdl_dynamic_resource_parameters_s;
 
 /**
- * \brief Defines parameters for the resource.
+ * \brief Defines static parameters for the resource.
  */
-typedef struct sn_nsdl_resource_info_ {
-
-    unsigned int                    mode:2;                     /**< STATIC etc.. */
-
-    unsigned int                    access:4;
-
-    bool                            publish_uri:1;
-
-    bool                            is_put:1; //if true, pointers are assumed to be consts (never freed). Note: resource_parameters_ptr is always freed!
-
-    uint8_t                         external_memory_block;
-
-    uint16_t                        pathlen;                    /**< Address */
-
-    uint16_t                        resourcelen;                /**< 0 if dynamic resource, resource information in static resource */
-
-    sn_nsdl_resource_parameters_s   *resource_parameters_ptr;
-
-    uint8_t (*sn_grs_dyn_res_callback)(struct nsdl_s *, sn_coap_hdr_s *, sn_nsdl_addr_s *, sn_nsdl_capab_e);
-
-    uint8_t                         *path;
-
-    uint8_t                         *resource;                  /**< NULL if dynamic resource */
+typedef struct sn_nsdl_static_resource_parameters_ {
 
-    ns_list_link_t                  link;
-} sn_nsdl_resource_info_s;
+    bool                                    publish_uri:1;
+    uint8_t                                 mode:2;                     /**< STATIC etc.. */
+    uint8_t                                 observable:2;
+    int16_t                                 pathlen;                    /**< Address */
+    uint16_t                                resourcelen;                /**< 0 if dynamic resource, resource information in static resource */
+    // Should this be other way around dynamic resource owns the static resource
+    sn_nsdl_dynamic_resource_parameters_s   *dynamic_resource_parameters;
```

Comment:
This can't work this way. This forces one to pre-allocate the 1:1 static sn_nsdl_dynamic_resource_parameters_s object. If the link went from sn_nsdl_dynamic_resource_parameters_s to sn_nsdl_static_resource_parameters, one could allocate the memory dynamically from heap when needed.